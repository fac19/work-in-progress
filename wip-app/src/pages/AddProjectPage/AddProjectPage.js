import React from "react";
import { Button } from "@material-ui/core";
import { AddProjectSection, AddProjectForm } from "./AddProjectPage.style";
import { useHistory } from "react-router-dom";
import { postAddProject } from "../../utils/post-fetch";

const AddProjectPage = (props) => {
  const history = useHistory();

  const handleSubmit = (event) => {
    event.preventDefault();

    const form = document.querySelector("form");
    const projectData = new FormData(form);

    postAddProject({
      project_name: projectData.get("project_name"),
      project_description: projectData.get("project_description"),
    })
      .then(() => history.push("/profile"))
      .catch((error) => console.error(error));
  };
  return (
    <AddProjectSection>
      <h1>Add New Project</h1>
      <AddProjectForm onSubmit={handleSubmit}>
        <label htmlFor="project_name">Project Name</label>
        <input
          type="text"
          id="project_name"
          name="project_name"
          required
        ></input>
        <label htmlFor="project_description">Project Description</label>
        <input
          type="text"
          id="project_description"
          name="project_description"
          required
        ></input>
        <Button variant="contained" color="primary" type="submit">
          Publish
        </Button>
      </AddProjectForm>
    </AddProjectSection>
  );
};

export default AddProjectPage;
