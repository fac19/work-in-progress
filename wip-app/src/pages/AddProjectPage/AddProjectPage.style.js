import styled from "styled-components";

const AddProjectSection = styled.section`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const AddProjectForm = styled.form`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 20rem;
`;

export { AddProjectSection, AddProjectForm };
