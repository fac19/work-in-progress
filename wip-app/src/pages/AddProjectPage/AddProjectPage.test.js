import React from "react"
import { render, screen, fireEvent } from "@testing-library/react"
import AddProjectPage from "./AddProjectPage"

test("Jest is working", () => {
  expect(true).toBeTruthy()
})
