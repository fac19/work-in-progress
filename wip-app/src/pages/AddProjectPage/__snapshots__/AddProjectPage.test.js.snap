// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test renders the correct ui 1`] = `
<section
  className="sc-AxjAm lcVokx"
>
  <h1>
    Add New Project
  </h1>
  <form
    className="sc-AxirZ kexMxH"
    onSubmit={[Function]}
  >
    <label
      className="sc-AxhCb djpVYo"
      htmlFor="project_name"
    >
      Project Name
    </label>
    <input
      className="sc-AxiKw dxRHDb"
      id="project_name"
      name="project_name"
      onChange={[Function]}
      placeholder="Project name..."
      required={true}
      type="text"
    />
    <label
      className="sc-AxhCb djpVYo"
      htmlFor="project_description"
    >
      Project Description
    </label>
    <input
      className="sc-AxiKw dxRHDb"
      id="project_description"
      name="project_description"
      onChange={[Function]}
      placeholder="Project description..."
      required={true}
      type="text"
    />
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Publish
      </span>
    </button>
  </form>
</section>
`;
